examples/ui-demo/src/hooks/useToast.tsx

```diff
@@ -0,0 +1,9 @@
+"use client";
+
+import { ToastContext } from "@/contexts/ToastProvider";
+import { useContext } from "react";
+
+export const useToast = () => {
+  const { toast, setToast, closeToast } = useContext(ToastContext);
```

Comment:
Can we throw an error if the context comes back as undefined from `useContext`?

Something like 
```
if (!context) { 
   throw Error("useToast can only be used by components that are descendants of the ToastProvider`");
}
```

Reply:
And then you can just return context no need to break it down and then send it a new object.